src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -296,9 +296,14 @@ def step(
                 )
 
             if variance_noise is None:
-                variance_noise = torch.randn(model_output.shape, dtype=model_output.dtype, generator=generator).to(
-                    device
-                )
+                if str(device) == "mps":
```

Comment:
This is not always correct, the device is usually `"mps:0"` in my tests. I'd rather use `device.type` when we can. But since schedulers accept either a `str` or a device argument, I'd do it like this:

```suggestion
                if str(device).startswith("mps"):
```

Reply:
See also: https://github.com/huggingface/diffusers/pull/1169#discussion_r1015369992

Reply:
Aah, good catch! The `device` here is not passed but derived from the device of `model_output`, so no need to use `str` here for "cpu". Changed to `torch.device("cpu")` so we could do `device.type`.